FEATURE(redis): Allow to use AOF

cdelgehier opened the pull request:
##### SUMMARY

By default Redis asynchronously dumps the dataset on disk.
This mode is good enough, but an issue with the Redis process may result into a few minutes of writes lost (depending on the configured save points).

The Append Only File is an alternative persistence mode that provides much better durability.

AOF and RDB persistence can be enabled at the same time without problems.

 ##### IMPACT
N/A

 ##### ADDITIONAL INFORMATION
OS-471

cdelgehier committed:
FEATURE(redis): Allow to use AOF

 ##### SUMMARY

By default Redis asynchronously dumps the dataset on disk.
This mode is good enough, but an issue with the Redis process may result into a few minutes of writes lost (depending on the configured save points).

The Append Only File is an alternative persistence mode that provides much better durability.

AOF and RDB persistence can be enabled at the same time without problems.

 ##### IMPACT
Yes.

For centos 7 systems, the AOF feature requires redis 4
```console
[openio@63e97cf1a446 ~]$ /usr/bin/redis-server /etc/oio/sds/TRAVIS/redis-0/redis.conf --daemonize no
*** FATAL CONFIG FILE ERROR ***
Reading the configuration file, at line 33
>>> 'aof-use-rdb-preamble yes'
Bad directive or wrong number of arguments
```

AOF is enabled by default in `19.10`

 ##### ADDITIONAL INFORMATION
OS-471

cdelgehier commented:
```
TASK [role_under_test : set slave status on bootstrap] *************************
Tuesday 14 January 2020  14:02:31 +0000 (0:00:00.045)       0:01:44.863 ******* 
FAILED - RETRYING: set slave status on bootstrap (5 retries left).
FAILED - RETRYING: set slave status on bootstrap (4 retries left).
FAILED - RETRYING: set slave status on bootstrap (3 retries left).
FAILED - RETRYING: set slave status on bootstrap (2 retries left).
FAILED - RETRYING: set slave status on bootstrap (1 retries left).
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ConnectionError: Error 111 connecting to 172.17.0.4:6011. Connection refused.
fatal: [localhost]: FAILED! => changed=false 
  attempts: 5
  msg: 'unable to connect to database: Error 111 connecting to 172.17.0.4:6011. Connection refused.
```
![bitmoji](https://render.bitstrips.com/v2/cpanel/c3bd1c53-57b5-43b0-9b0a-444ceb41ad35-0c2e4917-c514-46b4-81aa-9a122eb9bad8-v1.png?transparent=1&palette=1&width=246)

fatpat commented on README.md:
```diff
@@ -16,7 +16,15 @@ An Ansible role for redis. Specifically, the responsibilities of this role are t
 | `openio_redis_auth_pass` | `""` | Set the password to use to authenticate with the master and slaves |
+| `openio_redis_aof_load_truncated` | `"no"` |  Allow Redis to start without fix AOF file with `redis-check-aof`|
+| `openio_redis_aof_rewrite_incremental_fsync` | `"yes"` |  the AOF file will be fsync-ed every 32 MB of data generated |
+| `openio_redis_aof_use_rdb_preamble` | `"yes"` | Allow Redis to use an RDB preamble in the AOF file for faster rewrites and recoveries |
```
dispo uniquement à partir de `redis >= 4.0`

cdelgehier committed:
CHANGE(redis): Change for ansible 2.9 and drop of xenial

 ##### SUMMARY

Update .travis.yml according to our new requirements.

Update meta/main.yml too.

 ##### IMPACT
N/A

 ##### ADDITIONAL INFORMATION

fatpat commented on README.md:
```diff
@@ -16,18 +16,27 @@ An Ansible role for redis. Specifically, the responsibilities of this role are t
 | `openio_redis_bind_interface` | `"{{ ansible_default_ipv4.alias }}"` | The interface that this redis instance will run on |
 | `openio_redis_databases` | `16` | Set the number of databases |
-| `openio_redis_down_after` | `1000` | Number of milliseconds the master (or any attached slave or sentinel) should be unreachable |
+| `openio_redis_down_after` | `5000` | Number of milliseconds the master (or any attached slave or sentinel) should be unreachable |
```
why to rise the timeout from 1s to 5s ?

sebastienlapierre reviewed (approved):
Well done

fatpat reviewed (changes requested):
The `inpact` section of your commit message should not be `N/A` as this PR changes the default behavior by enabling `ADB`. Same applies for the timeout change which rise from 1s to 5s. I don't think it's a good idea to change it or at least write a warning about it in the commit message.

plus my comments on the code:
- some options requires redis >= 4 (how is it handled ?)
- why raising the default timeout from 1s to 5s ? 1s was fine to me but 5s seems a bit high

cdelgehier committed:
CHANGE(redis): Set `down-after-milliseconds` from 1s to 5s.

 ##### SUMMARY

The `down-after-milliseconds` value is 5000 milliseconds, that is 5 seconds, so masters will be detected as failing as soon as we don't receive any reply from our pings within this amount of time.

 ##### IMPACT
Yes.

 ##### ADDITIONAL INFORMATION
CUS-94
1s was too short, and we reporte several `odown` by day.
